src/Traits/HasPermissions.php

```diff
@@ -412,9 +412,11 @@ function ($object) use ($permissions, $model) {
      */
     public function syncPermissions(...$permissions)
     {
+        $this->collectPermissions(...$permissions);
```

Comment:
Should this be `$permissions = $this->collectPermissions($permission);`?
- assignment of returned value
- and drop the `...` when passing the parameter (also in line 419)

Also in `HasRoles`?

Reply:
I did think of that, but `collectPermissions` returns an array of ids as keys, example
```php
[
   'id_1' => [], 'id_1' => [], 'id_1' => []
]
```
If i do `$permissions = $this->collectPermissions($permission);` i have to add `array_keys`
```php
public function syncPermissions(...$permissions)
    {
        $permissions = $this->collectPermissions(...$permissions);

        $this->permissions()->detach();

        return $this->givePermissionTo(...array_keys($permissions));
    }

```
Still, that doesn't avoid double `collectPermissions` call


@drbyte **should i make the change?**

Reply:
Done

Reply:
> value is actually a collection, then when we assign the result and then have to run array_keys against it we lose that, don't we?

Yes, in that specific case.

>So, should we revert that $permissions =  part?

I am open to change what is needed to make it work in the best way, I do?

Reply:
Yes let's revert that, and the array_keys.
My apologies for the extra work!

Reply:
Don't worry, done